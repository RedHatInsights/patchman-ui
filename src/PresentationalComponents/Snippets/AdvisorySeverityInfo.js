import { Text, TextContent, TextVariants, Popover } from '@patternfly/react-core';
import { ExternalLinkSquareAltIcon } from '@patternfly/react-icons';
import propTypes from 'prop-types';
import React from 'react';
import { Flex, FlexItem } from '@patternfly/react-core';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';

const AdvisorySeverityInfo = ({ severity }) => {
    return (
        <Popover
            position="bottom"
            enableFlip
            headerContent={<div>{severity.label + ' severity'}</div>}
            bodyContent={
                <TextContent>
                    <Text component={TextVariants.p}>{severity.text}</Text>
                </TextContent>
            }
            footerContent={
                <a
                    href="https://access.redhat.com/security/updates/classification/"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    <Flex>
                        <FlexItem spacer={{ default: 'spacerSm' }}>
                            <ExternalLinkSquareAltIcon />
                        </FlexItem>
                        <FlexItem spacer={{ default: 'spacerSm' }}>
                            {intl.formatMessage(messages.linksSearchSecurityRatings)}
                        </FlexItem>
                    </Flex>
                </a>
            }
        >
            <a>{intl.formatMessage(messages.linksLearnMore)}</a>
        </Popover>
    );
};

AdvisorySeverityInfo.propTypes = {
    severity: propTypes.object
};

export default AdvisorySeverityInfo;
