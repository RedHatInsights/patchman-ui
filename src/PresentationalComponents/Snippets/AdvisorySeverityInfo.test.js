import AdvisorySeverityInfo from './AdvisorySeverityInfo';
import toJson from 'enzyme-to-json';

const severity = { label: 'testLabel', text: 'testText' };
const wrapper = shallow(<AdvisorySeverityInfo severity = {severity} />);

describe('AdvisorySeverityInfo', () => {
    it('Should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should redirect to securities app', () => {
        const hrefTag = wrapper.find('Popover').props().footerContent;
        const hrefProp = hrefTag.props.href;
        expect(hrefProp).toEqual('https://access.redhat.com/security/updates/classification/');
    });
});
