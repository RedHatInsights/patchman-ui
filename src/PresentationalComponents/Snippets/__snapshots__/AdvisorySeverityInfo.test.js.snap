// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvisorySeverityInfo Should match the snapshot 1`] = `
<Popover
  bodyContent={
    <TextContent>
      <Text
        component="p"
      >
        testText
      </Text>
    </TextContent>
  }
  enableFlip={true}
  footerContent={
    <a
      href="https://access.redhat.com/security/updates/classification/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="icon-with-label"
      >
        <ExternalLinkSquareAltIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
        Learn more about security ratings
      </span>
    </a>
  }
  headerContent={
    <div>
      testLabel severity
    </div>
  }
  position="bottom"
>
  <a>
    Learn more
  </a>
</Popover>
`;
